### PR TITLE
Fix package

### PR DESCRIPTION
**Pull request by syohex**
- Use cl-lib instead of cl
- Add autoload cookie for lazy loadyng
- Specify parent group
- Fix some byte-compile warnings
- Fix package URL
- Correct minimum Emacs version

There are some byte-compile warnings in original code.

```
tldr.el:34:1:Warning: cl package required at runtime
tldr.el:45:1:Warning: defgroup for ‘tldr’ fails to specify containing group
tldr.el:136:1:Warning: Lexical argument shadows the dynamic variable system-name

In tldr-get-file-path-from-command-name:
tldr.el:139:58:Warning: ‘system-name’ is an obsolete variable (as of 25.1);
    use (system-name) instead
tldr.el:142:1:Warning: Unused lexical variable ‘file-path’
```

### CODE REVIEW

**Comment by kuanyui** [comment]
Oh my... You are the best!!! Thank you!!!
